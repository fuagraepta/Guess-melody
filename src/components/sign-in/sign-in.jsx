import React from 'react';
import {PureComponent} from 'react';
import {withRouter} from 'react-router-dom';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {Operation} from '../../reducer/user/user-action-creator.js';
import {getUser} from '../../reducer/user/selectors.js';

class SignIn extends PureComponent {
  constructor(props) {
    super(props);

    this._loginRef = React.createRef();
    this._passwordRef = React.createRef();

    this._submitHandler = this._submitHandler.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (this.props.user !== prevProps.user) {
      const {history} = this.props;
      history.push(`/`);
    }
  }

  render() {
    return (
      <section className="login">
        <div className="login__logo"><img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83"/></div>
        <p className="login__text">Представтесь!</p>
        <form
          className="login__form"
          action=""
          onSubmit = {(evt) => {
            evt.preventDefault();
            this._submitHandler();
          }}>
          <p className="login__field">
            <label className="login__label" htmlFor="name">Логин</label>
            <input
              className="login__input"
              type="email"
              name="name"
              id="name"
              ref = {this._loginRef}
              required />
          </p>
          <p className="login__field">
            <label className="login__label" htmlFor="password">Пароль</label>
            <input
              className="login__input"
              type="password"
              name="password"
              id="password"
              ref = {this._passwordRef}
              required />
            <span className="login__error">Неверный пароль</span>
          </p>
          <button
            className="login__button button"
            type="submit">
            Войти
          </button>
        </form>
      </section>
    );
  }

  _submitHandler() {
    const {login} = this.props;
    const userData = {
      email: this._loginRef.current.value,
      password: this._passwordRef.current.value
    };

    login(userData);
  }
}

SignIn.propTypes = {
  login: PropTypes.func.isRequired,
  user: PropTypes.object,
  history: PropTypes.shape(
      {
        push: PropTypes.func
      }
  )
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  user: getUser(state)
});

const mapDispatchToProps = (dispatch) => ({
  login: (userData) => dispatch(Operation.login(userData))
});

export {SignIn};
export default connect(mapStateToProps, mapDispatchToProps)(withRouter(SignIn));
