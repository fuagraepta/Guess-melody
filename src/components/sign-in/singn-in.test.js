import React from 'react';
import renderer from 'react-test-renderer';

import {SignIn} from './sign-in.jsx';

describe(`The component is rendered correctly.`, () => {
  it(`Sign in screen is renders correctly.`, () => {
    const signIn = renderer
      .create(
          <SignIn
            login = {jest.fn()} />
      )
      .toJSON();

    expect(signIn).toMatchSnapshot();
  });
});
