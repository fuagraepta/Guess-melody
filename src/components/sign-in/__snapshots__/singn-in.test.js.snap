// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component is rendered correctly. Sign in screen is renders correctly. 1`] = `
<section
  className="login"
>
  <div
    className="login__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <p
    className="login__text"
  >
    Представтесь!
  </p>
  <form
    action=""
    className="login__form"
    onSubmit={[Function]}
  >
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="name"
      >
        Логин
      </label>
      <input
        className="login__input"
        id="name"
        name="name"
        required={true}
        type="email"
      />
    </p>
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="password"
      >
        Пароль
      </label>
      <input
        className="login__input"
        id="password"
        name="password"
        required={true}
        type="password"
      />
      <span
        className="login__error"
      >
        Неверный пароль
      </span>
    </p>
    <button
      className="login__button button"
      type="submit"
    >
      Войти
    </button>
  </form>
</section>
`;
